emscripten.py

```diff
@@ -1313,6 +1313,14 @@ def emscript_wasm_backend(infile, settings, outfile, libraries=None, compiler_en
     backend_args += ['-thread-model=single'] # no threads support in backend, tell llc to not emit atomics
     # disable slow and relatively unimportant optimization passes
     backend_args += ['-combiner-alias-analysis=false', '-combiner-global-alias-analysis=false']
+
+    # asm.js-style exception handling
+    if settings['DISABLE_EXCEPTION_CATCHING'] != 1:
+      backend_args += ['-enable-emscripten-cxx-exceptions']
+    if settings['DISABLE_EXCEPTION_CATCHING'] == 2:
```

Comment:
If we only add -emscripten-cxx-exceptions-whitelist only if len(settings['EXCEPTION_CATCHING_WHITELIST']) > 0, the whitelist set will be empty and all functions will be exception-enabled, which is the opposite of the intended behavior. I guess maybe we should change the fake function name to '__fake' or something to prevent collision in case the user code has 'fake' function too.

Actually, this snippet of code was copied from the existing code here, so if we are changing this we need to change this part too.
https://github.com/aheejin/emscripten/blob/91bc2f16bdf205ea6327ba9e9522d9546e191a15/emscripten.py#L126